client-core/src/wallet.rs

```diff
@@ -10,21 +10,29 @@ use zeroize::Zeroize;
 
 use chain_core::init::address::RedeemAddress;
 
-use crate::service::{KeyService, WalletService};
-use crate::{ErrorKind, PublicKey, Result, Storage};
+use crate::service::{BalanceService, KeyService, WalletService};
+use crate::{Chain, ErrorKind, PublicKey, Result, Storage};
 
 /// Interface for a generic wallet
-pub trait Wallet<K, W>
+pub trait Wallet<C, K, W, B>
 where
+    C: Chain,
     K: Storage,
     W: Storage,
+    B: Storage,
 {
+    /// Returns assiciated Crypto.com Chain client
+    fn chain(&self) -> &C;
+
     /// Returns associated key service
     fn key_service(&self) -> &KeyService<K>;
 
     /// Returns associated wallet service
```

Comment:
As above, these comments may better be deleted, as they don't add any information, or rephrased if there is some extra information.

for example, some of these services are used inside other services, so why are they exposed individually -- if all their functionality is only used inside a different service, why are they exposed?

Reply:
These comments are intended to be use with `cargo doc` command to generate API documentation. Also, these are exposed because these are required functions for any implementations of `Wallet` trait. Currently, `SledWallet` is the default implementation of this trait.

Reply:
The point of comments is to provide some additional information that's not immediately obvious from the name or types -- if `cargo doc` would complain about it, it could be suppressed with a justification comment.